### PR TITLE
Shared Image Gallery QuickStart Templates

### PR DESCRIPTION
**Pull request by axayjo**
Shared Image Gallery QuickStart Templates

### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by bmoore-msft** [comment]
@axayjo - check the CI failures, might be easier to break the PRs into on per sample while debugging...

**Comment by singhkays** [comment]
> @axayjo - check the CI failures, might be easier to break the PRs into on per sample while debugging...

@bmoore-msft This is a private preview service, CI sub is not registered to use this atm

File: 101-image-definition/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Create Image Version for an Image Definition
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-image-version%2Fazuredeploy.json" target="_blank">
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-image-definition/README.md]: Done

File: 101-image-definition/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-image-version%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template creates an Image Version . Please ensure you have deployed the 101 templates to create the Shared Image Gallery and Image Definition first.
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-image-definition/README.md]: Fixed

File: 101-image-definition/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Create Image Version for an Image Definition
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-image-version%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-image-version%2Fazuredeploy.json" target="_blank">
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-image-definition/README.md]: Done

File: 101-image-definition/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+	"galleryName": {
+			"type": "String",
+            "metadata": {
+                "description": "Name of the Shared Image Gallery."
+        }
+    },
+	"galleryImageName": {
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-image-definition/azuredeploy.json]: Done

File: 101-image-version/README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Hub and Spoke Topology Sandbox
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-image-version/README.md]: Fixed

File: 101-image-version/README.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-hub-and-spoke-sandbox%2Fazuredeploy.json" target="_blank">
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-image-version/README.md]: Done

File: 101-image-version/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,61 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+	"sourceManagedImage":	{
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-image-version/azuredeploy.json]: Done

File: 101-image-version/README.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-hub-and-spoke-sandbox%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template creates a basic hub-and-spoke topology setup. It creates a Hub VNet with subnets DMZ, Management, Shared and Gateway (optionally), with two Spoke VNets, development and production, containing a workload subnet each. It also deploys a Windows Jump-Host on the Management subnet of the HUB, and establishes VNet peerings between the Hub and the two spokes.
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-image-version/README.md]: Done

File: 101-image-version/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+			"description": "Resource ID of the source managed image."
+		}
+	}, 
+	"galleryName": {
+			"type": "String",
+            "metadata": {
+                "description": "Name of the Shared Image Gallery."
+        }
+    },
+	"galleryImageName": {
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-image-version/azuredeploy.json]: Done

File: 101-shared-image-gallery/README.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
+# Create Shared Image Gallery
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-shared-image-gallery%2Fazuredeploy.json" target="_blank">
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-shared-image-gallery/README.md]: Done

File: 101-shared-image-gallery/create-image-version.json [hunk 11]
```diff
@@ -0,0 +1,43 @@
+{
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-shared-image-gallery/create-image-version.json]: Deleted

File: 101-vm-from-image-version/README.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
+# Create Virtual Machine from an Image Version
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-vm-from-image-version/README.md]: Done

File: 101-vm-from-image-version/README.md [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-shared-image-gallery%2Fazuredeploy.json" target="_blank">
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-vm-from-image-version/README.md]: Done

File: 101-vm-from-image-version/azuredeploy.json [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+                "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+            }
+        },
+		"galleryName": {
+			"type": "String",
+            "metadata": {
+                "description": "Name of the Shared Image Gallery."
+            }
+        },
+		"galleryImageName": {
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-vm-from-image-version/azuredeploy.json]: Done

File: 101-vm-from-image-version/azuredeploy.json [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+        "subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnetName'))]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Storage/storageAccounts",
+            "sku": {
+                "name": "Standard_LRS"
+            },
+            "kind": "Storage",
+            "name": "[variables('storageAccountName')]",
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-vm-from-image-version/azuredeploy.json]: Removed

File: 101-vm-from-image-version/azuredeploy.parameters.json [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+  "parameters": {
+    "adminUsername": {
+      "value": "GEN-UNIQUE"
+    },
+    "adminPassword": {
+      "value": "GEN-PASSWORD"
+    },
+    "dnsLabelPrefix": {
+      "value": "GEN-UNIQUE"
+    }
```
[the remark this answers is not in the document]
Comment (axayjo) [reply 101-vm-from-image-version/azuredeploy.parameters.json]: Done

**Comment by singhkays** [comment]
#lgtm

**Comment by nagarajanpv** [comment]
I am trying to create a VM using an Image residing in Shared Image Gallery. I am using the template in the same way you had given. But I am not able to create the VM and is throwing error. Can you please help me in providing the sample template to create a VM form Shared Image Gallery

**Pull request merged**